src/moleculeSimulator/Simulator.cpp

```diff
@@ -4,73 +4,132 @@
+            model = std::make_unique<DirectSum>(*force, simulationSettings.parametersDirectSum.deltaT, inputFormat,
+                                                outputFormat);
+        }
+        break;
+        case TypeOfModel::linkedCells: {
+            switch (simulationSettings.parametersLinkedCells.force) {
+                case TypeOfForce::gravity: {
+                    force = std::make_unique<Gravity>();
+                }
+                break;
+                case TypeOfForce::leonardJonesForce: {
+                    force = std::make_unique<LeonardJonesForce>(simulationSettings.parametersLinkedCells.epsilon,
+                                                                simulationSettings.parametersLinkedCells.sigma);
+                }
+                break;
+                default: {
+                    spdlog::error("Use of invalid force type! Programm will be terminated!");
+                    exit(-1);
+                }
+            }
+            deltaT = simulationSettings.parametersLinkedCells.deltaT;
+            endT = simulationSettings.parametersLinkedCells.endT;
+            model = std::make_unique<LinkedCells>(*force, simulationSettings.parametersLinkedCells.deltaT,
+                                                  simulationSettings.parametersLinkedCells.domainSize,
+                                                  simulationSettings.parametersLinkedCells.rCutOff,
+                                                  simulationSettings.parametersLinkedCells.sigma, inputFormat,
+                                                  outputFormat,
+                                                  simulationSettings.parametersLinkedCells.boundarySettings);
+        }
+        break;
+        default: {
+            spdlog::error("Use of invalid model type! Programm will be terminated!");
+            exit(-1);
         }
-    }
-}
-
-void Simulator::calculateF() {
-    //Reset all forces and save current forces in old force variables.
-    for (auto &p: particles) {
-        p.setOldF(p.getF());
-        p.setF({0, 0, 0});
     }
 
-    //Iterate over all distinct pairs of particles and apply Newtons third law of motion.
+    //Add particles and objects of particles
 
-    for (auto p_i = particles.begin(); p_i != particles.end(); std::advance(p_i,1)) {
-        for (auto p_j = std::next(p_i); p_j != particles.end(); std::advance(p_j,1)) {
-            auto f_ij{force.compute(*p_i, *p_j)};
-            p_i->setF(p_i->getF() + f_ij);
-            p_j->setF(p_j->getF() - f_ij);
-        }
+    //Particles
+    for (auto pT: simulationSettings.particles) {
+        Particle p{pT.x, pT.v, pT.m};
+        model->addParticle(p);
     }
-}
-
-void Simulator::calculateX() {
-    for (auto &p: particles) {
-        p.setX(p.getX() + deltaT * p.getV() + ((deltaT * deltaT) / (2.0 * p.getM())) * p.getOldF());
+    //Cuboids
+    for (auto cuboid: simulationSettings.cuboids) {
```

Comment:
```suggestion
    for (auto &cuboid: simulationSettings.cuboids) {
```
Avoids some unnecessary copies. Of course, doesn't make a big difference here.